### PR TITLE
Couchdb Metrics Receiver

### PR DESCRIPTION
**Pull request by JonathanWamsley**
Add support for Couchdb Metrics. 

NOTE: SLES is not supported natively by Couchdb.

### CODE REVIEW

**Comment by JonathanWamsley** [comment]
The ops agent integration test had an unrelated failure on TestLoggingAgentCrashRestart failed failed sles test

File: confgenerator/testdata/valid/linux/metrics-receiver_couchdb/input.yaml
Comment (qingling128) [inline confgenerator/testdata/valid/linux/metrics-receiver_couchdb/input.yaml:22]: We can remove the "collection_interval" field from here.

For the context, there should at least be one valid unit test that verifies the absolute minimum user configuration. All optional configurations should be absent for this case including "collection_interval".

File: docs/couchdb.md
Comment (qingling128) [inline docs/couchdb.md:27]: `collection_interval` is no longer a required field. We can remove it from the sample config.

File: docs/couchdb.md
Comment (qingling128) [inline docs/couchdb.md:24]: It's preferred for the receiver and pipeline IDs to stay the same as the receiver type in the sample configuration. E.g.  "s/couchdb_metrics/couchdb", "s/couchdb_pipeline/couchdb"

File: integration_test/third_party_apps_data/test_config.yaml
Comment (qingling128) [inline integration_test/third_party_apps_data/test_config.yaml:27]: The `submodules/collectd` and `submodules/fluent-bit` might have sneaked in accidentally. Diff for those 2 submodules is not required for this PR, right?

image

File: confgenerator/testdata/valid/linux/metrics-receiver_couchdb/input.yaml
Comment (JonathanWamsley) [reply confgenerator/testdata/valid/linux/metrics-receiver_couchdb/input.yaml]: updated

File: docs/couchdb.md
Comment (JonathanWamsley) [reply docs/couchdb.md]: updated

File: docs/couchdb.md
Comment (JonathanWamsley) [reply docs/couchdb.md]: updated

File: integration_test/third_party_apps_data/test_config.yaml
Comment (JonathanWamsley) [reply integration_test/third_party_apps_data/test_config.yaml:27]: fixed

File: docs/couchdb.md
Comment (quentinmit) [inline docs/couchdb.md:43]: These custom units are all `{requests}`, `{views}`, etc., right? (with the curly brackets)

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_couchdb
Comment (quentinmit) [inline integration_test/third_party_apps_data/agent/ops-agent/linux/enable_couchdb:24]: Please test the default values for these parameters (so no endpoint, collection interval, username, password, etc.)

File: integration_test/third_party_apps_data/applications/couchdb/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/couchdb/centos_rhel/install:33]: What happens if you use the default configuration instead?

File: integration_test/third_party_apps_data/test_config.yaml
Comment (qingling128) [reply integration_test/third_party_apps_data/test_config.yaml:27]: Thanks!

**Comment by qingling128** [comment]
Overall LGTM modulo the remaining 3 open threads are addressed, and the integration tests are fixed (Probably some minor issue triggered by the suggested renaming):

image

File: docs/couchdb.md
Comment (JonathanWamsley) [reply docs/couchdb.md]: fixed

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_couchdb
Comment (JonathanWamsley) [reply integration_test/third_party_apps_data/agent/ops-agent/linux/enable_couchdb]: Hey Quentin, the default configuration will not work. In order to curl the metrics endpoint, a user:password is required or else an `unauthorized` error is returned. Since the default admin user has no default password a [new admin](https://docs.couchdb.org/en/stable/intro/security.html#creating-a-new-admin-user) needs to be created which can be done in the local.ini file.

This also means that the enable_couchdb config will require a username and password too.

File: integration_test/third_party_apps_data/applications/couchdb/centos_rhel/install
Comment (JonathanWamsley) [reply integration_test/third_party_apps_data/applications/couchdb/centos_rhel/install:33]: see above

**Review by qingling128 — APPROVED** [review]
LGTM.

Needs rebasing before merging. If the conflicts are limited to the golden files, we can skip running the integration tests again, as they passed already.

File: docs/couchdb.md
Comment (qingling128) [inline docs/couchdb.md:2]: Note for Jimmy, the full documentation can be found in https://github.com/GoogleCloudPlatform/ops-agent/blob/90d604b692cffb9cf388948e586cf5e2a894b2d0/docs/couchdb.md

Note for future reference: Currently we are trying to automate the documentation generation for https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party. Before that is completed, the workflow is:
1. ObserveIQ contributes documentation (in the format of https://github.com/GoogleCloudPlatform/ops-agent/blob/90d604b692cffb9cf388948e586cf5e2a894b2d0/docs/couchdb.md) as `docs/{APP_NAME}.md` files.
2. Jimmy to take these markdown and turn them into the expected format for https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party
3. We then comes back to GitHub to delete the install instructions to avoid maintaining two copies for the long term after https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party/{APP_NAME} is published.

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_couchdb
Comment (qingling128) [reply integration_test/third_party_apps_data/agent/ops-agent/linux/enable_couchdb]: Thanks for the clarification. Looks like all parameters have been removed except username and password, which as explained are required.

Resolving this.

**Pull request merged**
